x/magpie/client/cli/tx.go

```diff
@@ -43,24 +43,25 @@ func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
 		Short: "create a new post",
 		Args:  cobra.ExactArgs(4),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
 			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
 
 			accGetter := authtypes.NewAccountRetriever(cliCtx)
-
 			from := cliCtx.GetFromAddress()
 			if err := accGetter.EnsureExists(from); err != nil {
 				return err
 			}
 
-			msg := types.NewMsgCreatePost(args[0], args[1], time.Now(), from, args[2], args[3])
-			var err = msg.ValidateBasic()
+			parentId, err := types.ParsePostId(args[1])
 			if err != nil {
 				return err
 			}
 
-			// cliCtx.PrintResponse = true
+			msg := types.NewMsgCreatePost(args[0], parentId, time.Now(), from, args[2], args[3])
```

Comment:
I'm wondering if we should keep posting the time from client. The time can be manipulated on the client's computer and make the event happens before the latest state.